File: frontend/lib/models/chatbot_log.dart
class ChatbotLog {
  final String id;
  final String inquiryId;
  final String customerQuery;
  final String botResponse;
  final bool escalated;
  final DateTime createdAt;

  ChatbotLog({
    required this.id,
    required this.inquiryId,
    required this.customerQuery,
    required this.botResponse,
    required this.escalated,
    required this.createdAt,
  });

  factory ChatbotLog.fromJson(Map<String, dynamic> json) {
    return ChatbotLog(
      id: json['_id'],
      inquiryId: json['inquiry_id'],
      customerQuery: json['customer_query'],
      botResponse: json['bot_response'],
      escalated: json['escalated'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}

File: backend_zip/backend/models/ChatbotLog.js
const mongoose = require('mongoose');

const chatbotLogSchema = new mongoose.Schema({
  inquiry_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Inquiry', required: true },
  customer_query: { type: String, required: true },
  bot_response: { type: String, required: true },
  escalated: { type: Boolean, default: false },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('ChatbotLog', chatbotLogSchema);
